### PR TITLE
test: add side navigation integration tests

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- add `jest` & `@testing-library/dom` dependencies and basic setup for integration tests with JSDOM
- Add side navigation tests

_Note:_ includes commits from https://github.com/canonical/vanilla-framework/pull/4536

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [ ] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: tests/integration/side-navigation.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const container = document.createElement('div');
+  container.innerHTML = html;
+  document.body.appendChild(container);
+  import(scriptUrl);
+  return {container};
+}
+
+it('adds aria-expanded to list elements on load', async () => {
+  const {container} = render(`
+  <ul class="p-side-navigation__list">
```
Comment (bartaz) [inline tests/integration/side-navigation.test.js:15]: Does this part of HTML need to be hardcoded? Or can it be included from some example?

File: jest-setup.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import '@testing-library/jest-dom';
```
Comment (bartaz) [inline jest-setup.js:1]: It it a common convention to put it in root folder or should/could it live somewhere else, like `/test` (although we have some CSS metrics tests there I think)

File: tests/integration/side-navigation.test.js (hunk 1 above)
Comment (petermakowski) [reply tests/integration/side-navigation.test.js:15]: Could be imported if you think that works better, added this as an alternative - it's quite nice to see it all in a single place.

File: jest-setup.js (hunk 2 above)
Comment (petermakowski) [reply jest-setup.js:1]: This file is usually placed in the root folder along with `package.json`, eslint config etc.

**Comment by petermakowski** [comment]
We decided to close this for now. We can get much more value from e2e tests: https://github.com/canonical/vanilla-framework/pull/4540

**Pull request closed without merging**